lib/local_daily_digest.js

```diff
@@ -0,0 +1,448 @@
+					checkedInUsers: checkInTemp,
+					checkedOutUsers: checkOutTemp
+				});
+
+			});
+			return cb();
+		},
+		function (cb){
+			sortMailer(finalOutput);
+		}		
+	]);
+}	
+
+function sortMailer(finalOutput){
+
+	var tempProfileTuple=[];
+	async.series([
+		//for each manager, get a list of the users checked in and checked out of the required locations and send them to the mailer
+		function(cb){
+			if(finalOutput.length != 0){
+				finalOutput.forEach(function(managerEditor){
+					if((managerEditor.checkedInUsers.length != 0 )|| (managerEditor.checkedOutUsers.length != 0)){
+						tempProfileTuple.push(managerEditor);
+					}
+				});
+			}		
+			else
+				return cb();
+			return cb();
+		},
+		function(cb){
+			tempProfileTuple.forEach(function(tuple){
+				mailTemplate(tuple);
+			});
+		}
+	]);
+}
+
+
+function mailTemplate(tuple){
+
+	var managerEmail  = '',
+		location=[],
+		currentCheckedInUser = [];
+		currentCheckedOutUser=[],
+		mailOptions ={},
+		sortedCheckIn =[],
+		sortedCheckOut = [],
+		target =[],
+		tempSortedCheckIn = [],
+		final = [],
+		org_out = [],
+		org_in =[],
+		tempSortedCheckOut = [];
+
+	managerEmail = tuple.managerID.userid.replace(/(_\d+)$/,'');
+	async.series([	
+		function (cb){
+
+			// group the checked in and checked out users by location
+			//Hence sending one email perlocation for the checkedin and checkedout users
+			tuple.managerID.dailyDigest.forEach(function(location){
+				tuple.checkedInUsers.forEach(function(user){
+					if(user.cIn_locationId === location){
+						currentCheckedInUser.push(user);
+					}
+				});
+
+				tuple.checkedOutUsers.forEach(function(user){
+					if(user.cOut_locationId === location){
+						currentCheckedOutUser.push(user);
+					}
+				});
+			});
+			return cb();
+		},
+		function(cb){
+
+			//edge case scenario, a user checks in, checks out and checks in again in less then 24 hours
+			//Hence has duplicate entries. Remove the duplicates
+			sortedCheckIn = _.filter(currentCheckedInUser, function (element, index) {
+			    for(index += 1; index < currentCheckedInUser.length; index += 1) {
+			        if (_.isEqual(element[0], currentCheckedInUser[index])) {
+			            return false;
+			        }
+			    }
+			    return true;
+			});
+	
+			sortedCheckOut = _.filter(currentCheckedOutUser, function (element, index) {
+			    for(index += 1; index < currentCheckedOutUser.length; index += 1) {
+			        if (_.isEqual(element[0], currentCheckedOutUser[index])) {
+			            return false;
+			        }
+			    }
+			    return true;
+			});
+
+			//sort out by location name
+			sortedCheckIn.sort(function(a, b) {
+			    return a.cIn_location.localeCompare(b.cIn_location);
+			});
+
+			sortedCheckOut.sort(function(a, b) {
+			    return a.cOut_location.localeCompare(b.cOut_location);
+			});
+			return cb();
+		},
+		function (cb){
+			var i= 0;
+
+			//group all required ID' names etc based on location and remove previous instances
```

Comment:
Couldn't this sorting happen above, when retrieving the users ? I believe you should have an array with, for each location where a daily digest needs to be sent out, the checkedin users and the checkedout users. Something like: locations[locationId].checkedIn and locations[locationId].checkedOut